src/overlays/actors/ovl_En_Raf/z_en_raf.c

```diff
@@ -80,65 +178,787 @@ static DamageTable D_80A1939C = {
+}
+
+/**
+ * Makes the trap petals on the lily pad dissolve and switches the lily pad to the "dead" state.
+ * When the water in Woodfall Temple is purified, this function and EnRaf_Convulse are jointly
+ * responsible for controlling the lily pad's death.
+ */
+void EnRaf_Dissolve(EnRaf* this, GlobalContext* globalCtx) {
+    f32 curFrame = this->skelAnime.curFrame;
+    s32 i;
+
+    if (this->endFrame <= curFrame) {
+        this->dissolveTimer++;
+        if (this->dissolveTimer < (BREG(3) + 105)) {
+            for (i = 0; i < (BREG(4) + 5); i++) {
+                EnRaf_ClearPixelsPetal(Lib_SegmentedToVirtual(&gCarnivorousLilyPadTrapPetalTex),
+                                       sPetalClearPixelTableFirstPass, this->petalClearPixelFirstPassIndex);
+                EnRaf_ClearPixelsTeeth(Lib_SegmentedToVirtual(&gCarnivorousLilyPadTrapTeethTex),
+                                       sTeethClearPixelTableFirstPass, this->teethClearPixelFirstPassIndex);
+                if (this->petalClearPixelFirstPassIndex < 0x200) {
+                    this->petalClearPixelFirstPassIndex++;
+                }
+
+                if (this->teethClearPixelFirstPassIndex < 0x40) {
+                    this->teethClearPixelFirstPassIndex++;
+                }
+            }
+        }
+    }
+
+    if (this->dissolveTimer > (BREG(5) + 50)) {
+        for (i = 0; i < (BREG(6) + 5); i++) {
+            EnRaf_ClearPixelsPetal(Lib_SegmentedToVirtual(&gCarnivorousLilyPadTrapPetalTex),
+                                   sPetalClearPixelTableSecondPass, this->petalClearPixelSecondPassIndex);
+            EnRaf_ClearPixelsTeeth(Lib_SegmentedToVirtual(&gCarnivorousLilyPadTrapTeethTex),
+                                   sTeethClearPixelTableSecondPass, this->teethClearPixelSecondPassIndex);
+            if (this->petalClearPixelSecondPassIndex < 0x200) {
+                this->petalClearPixelSecondPassIndex++;
+            }
+
+            if (this->teethClearPixelSecondPassIndex < 0x40) {
+                this->teethClearPixelSecondPassIndex++;
+            }
+        }
+    }
+
+    if (this->dissolveTimer > (BREG(7) + 160)) {
+        for (i = CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_LOWER_SEGMENT; i <= CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_UPPER_SEGMENT;
+             i++) {
+            Math_Vec3f_Copy(&this->limbScale[i], &gZeroVec3f);
+            Math_Vec3f_Copy(&this->targetLimbScale[i], &gZeroVec3f);
+        }
+
+        EnRaf_SetupDeadIdle(this);
+    }
+}
+
+void EnRaf_SetupDeadIdle(EnRaf* this) {
+    if (this->action == EN_RAF_ACTION_EXPLODE) {
+        this->timer = 90;
+    } else {
+        this->action = EN_RAF_ACTION_DEAD_IDLE;
+    }
+
+    this->actionFunc = EnRaf_DeadIdle;
+}
+
+/**
+ * Simply sits around doing nothing. If the revive timer is non-zero, then this function
+ * will decrement the revive timer and revive the trap petals once it reaches 0.
+ */
+void EnRaf_DeadIdle(EnRaf* this, GlobalContext* globalCtx) {
```

Comment:
I suppose there is a question of whether it can be dead if it gets better... how about "dormant"?